src/util/tab-events.test.js

```diff
@@ -0,0 +1,185 @@
+        expect(browser.tabs.executeScript).toHaveBeenCalledWith(tabId, {code: 'undefined', runAt: 'document_end'})
+    })
+
+    test('should reject the promise if the script is not executed', async () => {
+        browser.tabs = {
+            executeScript: jest.fn().mockReturnValue(
+                new Promise((resolve, reject) => {
+                    reject(new Error('Script unable to execute'))
+                })
+            ),
+        }
+        try {
+            await whenPageDOMLoaded({tabId})
+        } catch (err) {
+            expect(err.toString()).toBe('Error: Script unable to execute')
+        }
+    })
+
+    test('should resolve promise when DOM page is loaded', async () => {
+        expect.assertions(2)
+        browser.tabs = {
+            executeScript: jest.fn().mockReturnValue(
+                new Promise((resolve, reject) => {
+                    resolve()
+                })
+            ),
+        }
+        await whenPageDOMLoaded({tabId})
+        expect(browser.tabs.executeScript).toHaveBeenCalledWith(tabId, {code: 'undefined', runAt: 'document_end'})
+        expect(eventToPromise.default).toHaveBeenCalled()
+    })
+
+    test('should reject the promise if tab is changed', async () => {
+        browser.tabs = {
+            executeScript: jest.fn().mockReturnValue(
+                new Promise((resolve, reject) => {
+                    resolve()
+                })
+            ),
+        }
+        eventToPromise.default = jest.fn().mockReturnValue(
+            new Promise((resolve, reject) => {
+                reject(new Error('Tab was changed'))
+            })
+        )
+        try {
+            await whenPageDOMLoaded({tabId})
+        } catch (err) {
+            expect(err.toString()).toBe('Error: Tab was changed')
+        }
+    })
+})
+
+describe('whenPageLoadComplete', () => {
+    const tabId = 1
+
+    beforeAll(() => {
+        eventToPromise.default = jest.fn().mockReturnValue(
+            new Promise((resolve, reject) => {
+                resolve()
+            })
+        )
+        browser.webNavigation = {
+            onCommitted: jest.fn(),
+        }
+    })
+
+    test('should return if the tab status is complete', async () => {
+        expect.assertions(2)
+        browser.tabs = {
+            get: jest.fn().mockReturnValueOnce({
+                status: 'complete',
+            }),
+        }
+        await whenPageLoadComplete({tabId})
+        expect(browser.tabs.get).toHaveBeenCalledWith(tabId)
+        expect(eventToPromise.default).not.toHaveBeenCalled()
+    })
+
+    test('should return a promise which resolves', async () => {
+        browser.tabs = {
+            get: jest.fn().mockReturnValueOnce({
+                status: 'incomplete',
+            }),
+        }
+        await whenPageLoadComplete({tabId})
+        expect(eventToPromise.default).toHaveBeenCalled()
+    })
+
+    test('should reject a promise if tab is changed', async () => {
```

Comment:
'if tab is changed' is too generic, it could sound like e.g. one is activating another tab. Something like 'if tab has changed location' gets closer.